build/dsc/resource.rb

```diff
@@ -116,13 +116,18 @@ def has_credential?
 
     def ps_module
       unless @ps_module
+
         path_array = @resource_mof_path.split('/')
         revert_array = path_array.reverse
         downcased_array = revert_array.collect{|p| p.downcase}
         index = downcased_array.index('dscresources')
-        raise "module for #{self.name} not found" if index == nil
+        raise "module for #{self.name} not found (Missing DSCResources directory in path #{@resource_mof_path})" if index == nil
 
         module_name = revert_array[index + 1 ] rescue nil
+        if (module_name == "dsc_resources") 
+          warn("The name of the module in directory #{@resource_mof_path} was detected as #{module_name}.  This may indicate that the wrong directory was used to import the resource")
```

Comment:
I wanted to keep this PR fairly simple and note create additional functionality yet i.e. docs and error message improvements.  Ideally, the import process should be doing this detection, not the build process (fail fast).  I will raise an additional ticket to review the import/build process scripts to take into account more varied directory structures e.g. Right now, it is difficult to import a single StackExchange resource

(https://github.com/PowerShellOrg/StackExchangeResources/tree/master/DSCResources)